stablehlo/reference/InterpreterOps.td

```diff
@@ -80,4 +80,38 @@ def Interpreter_RunParallelOp : Op<Interpreter_Dialect, "run_parallel", []> {
   let hasVerifier = 1;
 }
 
+def Interpreter_ProbeOp : Op<Interpreter_Dialect, "probe",
+    [SameOperandsAndResultType]> {
+  let arguments = (ins
+    HLO_Tensor:$operand,
+    StrAttr:$probe_id
+  );
+  let results = (outs HLO_Tensor:$result);
+
+  let description = [{
+    Probe and store the values of the input tensor at runtime, using the NumPy
+    file format and using the `probe_id` string as the serialization filename,
+    storing results in the directory specified by the `--probe-output-dir`
+    flag.
```

Comment:
Add a little more detail on the I/O outputs of this op. Given that they are encoded in `evalProbeOp` they are a part of the semantics of the op. I'm realizing this description serves as the only documentation for this interpreter feature currently:

Something like the following (please correct any wording if I have it wrong):

"Writes tensor input value to `<output-dir>/<probe_id>_<iteration>.npy`. Additionally, adds an entry to `<output-dir>/index.csv` metadata file which maps probe IDs and iterations to filenames with their tensor values"